docs: clarify mutation ordering for MutatingAdmissionPolicy (#53977)
kubernetes/website: content/en/docs/reference/access-authn-authz/mutating-admission-policy.md

Only property names of the form `[a-zA-Z_.-/][a-zA-Z0-9_.-/]*` are accessible.

## Mutation ordering and intermediate state

When a MutatingAdmissionPolicy defines multiple mutations, each mutation is applied
independently to the object. Between mutations, the following occurs:

> - The object is serialized to JSON and then parsed back

Reviewer: For example, the API server might be using CBOR to serialize that resource. Or it might be an aggregated API server that serializes to XML because weirdness, but that still honors MutatingAdmissionPolicy.

- Built-in defaulting is applied

> - Empty fields (fields with `omitempty` that are absent or null) may be removed

Reviewer: ```suggestion
- Empty fields (fields defined as `omitempty`, and that are absent or null) may be removed
```

> Because of this behavior, a mutation must not assume that fields added by a previous

Reviewer: ```suggestion
Because of this behavior, a mutation **must not** assume that fields added by a previous
```

mutation will still exist when it runs. The intermediate state after each mutation
may differ from what would exist if all mutations were applied to a single in-memory object.

### Single mutation vs. multiple mutations

> A single mutation expression can contain multiple JSONPatch operations that are applied

Reviewer: ```suggestion
A single mutation expression can contain multiple JSON patch operations that are applied
```

There's a space, according to https://datatracker.ietf.org/doc/html/rfc6902

Alternatively, wrap JSONPatch in backticks:
```suggestion
A single mutation expression can evaluate to multiple `JSONPatch` operations, that are applied
```

in sequence to the same in-memory object. For example:

```cel
[
  JSONPatch{op: "add", path: "/spec/labels", value: Object.spec.labels{"app": "myapp"}},
  JSONPatch{op: "add", path: "/spec/labels/owner", value: "team-a"}
]
```

These two operations are applied to the same object, so the second operation can safely
depend on the result of the first.

In contrast, each expression in the `spec.mutations` list results in a separate mutation
that is applied independently. The intermediate object is serialized and deserialized
between mutations, which can cause fields to be lost.

### Best practices

> - **Group dependent operations**: If multiple JSONPatch operations depend on each other,

Reviewer: ```suggestion
- **Group dependent operations**: If multiple JSON Patch operations depend on each other,
```

  define them in a single CEL expression so they operate on the same in-memory object.

> - **Prefer ApplyConfiguration for list fields**: When mutating optional list fields,

Reviewer: ```suggestion
- **Prefer `ApplyConfiguration` for list fields**: When mutating optional list fields,
  `ApplyConfiguration` mutations use a [server-side apply](/docs/reference/using-api/server-side-apply/) patch strategy, which is
  safer for list manipulation than independent `JSONPatch` operations.
```
?
